kbfsgit/runner.go

```diff
@@ -1462,13 +1524,22 @@ func (r *runner) pushSome(
 			}
 			results[dst] = err
 		} else {
+			refName := plumbing.ReferenceName(refspec.Src())
```

Comment:
Have you tested this with wildcard pushes?  I don't actually know whether wildcard refspecs can come through the git-remote-helper interface, or if they get expanded elsewhere.  If the latter, can we error if `IsWildcard` here?  If the former, we probably need to find some gogit utility for expanding the wildcards.

Reply:
I haven't. Should we error, or should we just add the refspec and ignore the commits?

Reply:
We should test and see what happens when there's a wildcard on the command -- what does the remote helper see?  If it also sees a wildcard refspec, I think we have to support commits for it, it would be pretty common.

Reply:
OK I'll test.

Reply:
It looks like wildcards are ignored:
```
Johns-iMac.local 18:57:41 zipdroid (jzila/test-2) ➜ git push origin jzila/*
Initializing Keybase... done.
Syncing with Keybase... done.
Everything up-to-date
```
```
Johns-iMac.local 18:59:48 zipdroid (jzila/test-2) ➜ gp
Initializing Keybase... done.
Syncing with Keybase... done.
Counting and decrypting: 3 objects... done.
Preparing and encrypting: (100.00%) 3/3 objects... done.
Indexing hashes: (100.00%) 3/3 objects... done.
Indexing CRCs: (100.00%) 3/3 objects... done.
Indexing offsets: (100.00%) 3/3 objects... done.
Syncing encrypted data to Keybase: (100.00%) 41.92/41.92 KB... done.
Syncing encrypted data to Keybase: (100.00%) 18.72/18.72 KB... done.
To keybase://private/jzila/zipdroid.git
 * [new branch]      jzila/test-2 -> jzila/test-2
Branch 'jzila/test-2' set up to track remote branch 'jzila/test-2' from 'origin'.
```

Reply:
Actually I was using incorrect syntax. Looks like git expands the wildcard itself:
```
Johns-iMac.local 19:18:57 zipdroid (john) ➜ git push origin refs/heads/jzila/*
Initializing Keybase... done.
Syncing with Keybase... done.
Syncing encrypted data to Keybase: (100.00%) 20.39/20.39 KB... done.
Syncing encrypted data to Keybase: (100.00%) 18.72/18.72 KB... done.
To keybase://private/jzila/zipdroid.git
 * [new branch]      jzila/test-1 -> jzila/test-1
 * [new branch]      jzila/test-2 -> jzila/test-2
```

Which shows up in the logs as:
```
Johns-iMac.local 19:19:57 zipdroid (john) ➜ tail -f ~/Library/Logs/keybase.git.log | grep 'runner.go'
...
2018-01-03T19:20:10.163428-08:00 ▶ [DEBU git runner.go:1773] 3c8 Received command: push refs/heads/jzila/test-1:refs/heads/jzila/test-1
2018-01-03T19:20:10.163505-08:00 ▶ [DEBU git runner.go:1773] 3c9 Received command: push refs/heads/jzila/test-2:refs/heads/jzila/test-2
2018-01-03T19:20:10.163538-08:00 ▶ [DEBU git runner.go:1773] 3ca Received command:
2018-01-03T19:20:10.163595-08:00 ▶ [DEBU git runner.go:1704] 3cb Processing push batch [tags:GITID=Iavjs5KrdgI1yB_UwiXDmA,GITCMDID=GZFZBS8EIFbk1KRbpDmmyA]
```